File: csharp/Core/Cards/Smithy.cs
namespace Powercards.Core.Cards
{
    [CardInfo(CardSet.Base, 4, true)]
    public class Smithy : CardBase, IActionCard
    {
        #region methods
        public void Play(TurnContext context)
        {
            context.ActivePlayer.DrawCards(3, context);
        }
        #endregion
    }
}

File: java/powercards/src/main/java/powercards/cards/Smithy.java
package powercards.cards;

import powercards.ActionCard;
import powercards.Card;
import powercards.Cards;
import powercards.Game;

public class Smithy extends Card implements ActionCard {
  @Override
  public int getCost(Game game) {
    return 4;
  }

  @Override
  public void play(Game game) {
    Cards.drawCards(game.getActivePlayer(), 3, game.getDialog().inout());
  }
}
